plugin/entryBase.go

```diff
@@ -215,3 +256,29 @@ func (e *EntryBase) SetTestID(id string) {
 func notRunningTests() bool {
 	return flag.Lookup("test.v") == nil
 }
+
+var aliases = func() map[reflect.Type]*jsonschema.Type {
```

Comment:
jsonschema's a pretty good library, but there are some bugs. Since I want to get something out quick, I'm having Wash point to my fork of it containing hacky fixes for those bugs that work for our use-cases. The fork is at https://github.com/ekinanp/jsonschema. I've filed issues at https://github.com/alecthomas/jsonschema/issues for the things I had to fix in that fork.

Reply:
I wanted to use JSON schemas because they're a standard. `wash find` doesn't need the full-blown schema -- it only needs to know whether a given path (e.g. like `.tags[?].key`) is valid, so only the `properties` key for objects or the `items` keys for arrays are necessary.

The schemaP for the meta primary will consist of constructing a "mock" JSON object and then validating that object against the provided schema, returning true if validation succeeded, false otherwise. Thus, I'll be munging the JSON schema in `wash find` to remove things like `RequiredProperties` (since I only care that the properties exist), deleting `type` entries for the primitive types, and upcasing "property" names so that the matching key stuff works. Note that my reason for removing `type` for primitive types is b/c I don't think strict type-level validation is necessary for the meta primary, especially when we allow multiple types (string, int) for e.g. the time predicate. I think just validating that what the user types in leads to an existent metadata entry is more than enough.

On a side note, I'm wondering if I should change `JSONSchema` => `MetadataSchema` and just do the `wash find`-specific munging here so that the API can return it. Although that would require me to comment "This is for `wash find` only", which indicates that it belongs in `wash find`. I guess returning the schema as-is means we could re-use it if we ever need to for other things.

Reply:
I'm also wondering if `aliases` should be plugin-specific, maybe something provided by the plugin roots? Kube's `Time` type + `resource.Quantity` types are fairly common.

Reply:
We should work towards making those fixes in the upstream library. It'll be more maintainable in the long run.

Reply:
Aliases should definitely be something plugin-specific. Why would K8's time and quantity types be used outside of the kubernetes plugin?

Reply:
Ok. What do we do if the upstream library is not responsive? Do we write our own? Note that this was the only "good" library I could find that generated JSON schemas from Go structs.

Reply:
I guess we keep the fork if it's unresponsive. But it's worth having more than hacks, with testing, if we're going to rely on the library. And once that's done, it's easy to make an upstream PR.

Reply:
Ok. I can file an issue to implement a proper fix for the library once this is merged.